13-1-1/update_indicator.R

```diff
@@ -0,0 +1,387 @@
+  
+#
+# ons_deaths4 is deaths by COD by country by sex by year. 
+header_ons_4 <- which(ons_deaths4$X1 == "Country")
+names(ons_deaths4) <- ons_deaths4[header_ons_4,]
+ons_deaths4 <- ons_deaths4 %>% 
+  tail(-header_ons_4)
+
+# Find the columns with COD codes e.g. X32
+cause_of_death_bool <- grepl("^[X][0-9][0-9]$", names(ons_deaths4)) 
+cause_of_death_names <- names(ons_deaths4)[cause_of_death_bool]
+
+ons_deaths4 <- ons_deaths4 %>% 
+  select('Country','Sex','Year', cause_of_death_names) %>% 
+  tidyr::fill(c('Country', 'Sex')) %>% 
+  tidyr::pivot_longer(cols=cause_of_death_names, names_to="Cause of death", values_to = "Value") %>% 
+  mutate(Value = as.numeric(Value)) %>% 
+  mutate(Series = "Number of deaths from exposure to forces of nature") %>%
+  mutate(`Cause of death` = lapply(`Cause of death`, recode_disaster)) %>% 
+  mutate(`Unit multiplier` = "") %>%  # Blank column, multiplier is zero
+  mutate(Units = "Number") %>% 
+  mutate(`Observation status` = "Normal value") %>% 
+  mutate(Country = ifelse(Country=="England and wales", "", Country)) %>% 
+  mutate(GeoCode = lapply(Country, get_GeoCode)) %>% 
+  mutate(GeoCode = as.character(GeoCode)) %>% 
+  select(Year, Series, Country, Sex, `Cause of death`, `Observation status`, `Unit multiplier`, Units, GeoCode, Value) %>% 
+  mutate(Year = as.integer(Year)) %>%  #%>% # so we can keep only <2013
+  mutate(Value = as.numeric(Value)) %>% 
+  mutate(`Cause of death` = unlist(`Cause of death`)) # unlist or as.character?  
+
+# Now to calculate the values aggregated by sex
+ons_death4_sex_agg <- get_sex_aggregates_v4(ons_deaths4)
+
+ons_death4_sex_agg <- ons_death4_sex_agg %>% 
+  mutate(Series = "Number of deaths from exposure to forces of nature") %>% 
+  mutate(`Observation status` = "Normal value") %>% 
+  mutate(Sex = "") %>% 
+  mutate(Units = "Number") %>% 
+  mutate(`Unit multiplier` = "") %>% 
+  mutate(GeoCode = lapply(Country, get_GeoCode)) %>% 
+  mutate(GeoCode = as.character(GeoCode)) %>% 
+  select(Year, Series, `Cause of death`, Country, Sex, `Observation status`, `Unit multiplier`, `Units`, GeoCode, Value)
+
+# Combined all the ons data sets and their aggregations ----
+ons_combined <- ons_deaths1 %>% 
+  rbind(ons_deaths2) %>% 
+  rbind(ons_death2_sex_agg) %>% 
+  rbind(ons_deaths3) %>% 
+  rbind(ons_deaths4) %>% 
+  rbind(ons_death4_sex_agg)
+
+# Now to keep only the rows for Year<2012 as we will use the nomis data after
+ons_combined <- ons_combined %>% 
+  mutate(Year = as.numeric(Year)) %>% 
+  filter(Year <= 2012)
+
+# Now to process the nomis death data ----
+# Thankfully this data is already aggregated on Sex
+# Still need to aggregate on all cause of death
+
+
+nomis_deaths <- nomis_deaths %>%
+  rename(c(Sex = GENDER_NAME, `Cause of death` = CAUSE_OF_DEATH_NAME, Value = OBS_VALUE, Country = GEOGRAPHY_NAME, Year = DATE)) #%>% 
+
+nomis_deaths <- nomis_deaths %>%
+  mutate(`Observation status` = "Normal value") %>%
```

Comment:
you can do all the mutates together   
also the nomis data has observation status and geocode columns that it would be better to use rather than hard coding it e.g.
```
nomis_cleaned <- nomis_deaths %>%
    mutate(`Observation status` = OBS_STATUS_NAME,
            Geocode = GEOGRAPHY_CODE,
            Series = "Number of deaths from exposure to forces of nature",
            etc)

Reply:
Actioned